carbon-design-system/carbon
# fix(components): include side effects from src for rollup resolution

Closes #7773 

Adds in the relevant src files to the `sideEffects` array to correctly emit our document event listeners on boot.

#### Changelog

**New**

**Changed**

- Update `sideEffects` in `components/package.json` to include source files so that rollup's node-resolve plugin correctly emits imports that have side effects

**Removed**

#### Testing / Reviewing

- Checkout the PR
- Build the components package locally (`yarn gulp scripts:rollup` is a nice shortcut in `packages/components`)
- Check out `scripts/components.js` and verify that the following block of code appears at the end:

```js
  if (document.readyState === 'loading') {
    document.addEventListener('DOMContentLoaded', init);
  } else {
    // DOMContentLoaded has been fired already
    // Let consumer have chance to see if it wants automatic instantiation disabled, and then run automatic instantiation otherwise
    setTimeout(init, 0);
  }
```
